s3repo/model.py

```diff
@@ -378,9 +388,103 @@ def delete_package(self, package):
 
+    def is_dir_or_file(self, path):
+        """Check if path is directory or file."""
+
+        list_parameters = {'Bucket': self.bucket.name,
+                           'Prefix': path}
+
+        try:
+            result = self.s3_client.list_objects_v2(**list_parameters)
+        except self.s3_client.exceptions.NoSuchBucket:
+            raise RuntimeError("No such bucket.")
+        contents = result.get('Contents')
+        # if contents is not None:
+        #     subfolders = 0
+        #     for fileobj in contents:
+        #         key = contents.get('Key')
+        #         tail = key[len(path):]
+        #         # if tail
+            
+        #     if tail != '' and tail[0] == '/':
+        #         return 'dir'
+        return 'file'
+
+    def get_directory(self, path):
+        """Get lists and metadata of files and folders within
+        directory from S3.
+        """
+
+        paginator = self.s3_client.get_paginator('list_objects')
+        list_parameters = {'Bucket': self.bucket.name,
+                           'Prefix': path}
+
+        files = {}
+        folders = {}
+        try:
+            for result in paginator.paginate(**list_parameters):
+                if result.get('Contents') is not None:
+                    for fileobj in result.get('Contents'):
+                        # Select the names of all files and
+                        # folders inside the current one.
+                        directory_path_len = 0
+                        if len(path) != 0:
+                            directory_path_len = len(path)
+                        tail = fileobj.get('Key')[directory_path_len:]
+                        item_name = tail.split('/')[0]
+                        last_modified = fileobj.get('LastModified')
+
+                        # list_objects() paginator returns the
+                        # current directory, as well as files with
+                        # the same name (prefix), messing with the
+                        # current directory, because they are S3
+                        # objects too. Ignore these objects.
+                        if item_name == '':
+                            continue
+                        # If tail is longer than item_name (i.e.
+                        # have /... afetr item_name), it means
+                        # that we deal with directory.
+                        if tail != item_name:
+                            in_list = item_name in folders
+                            if (not in_list or in_list
+                                    and folders[item_name] < last_modified):
+                                folders[item_name] = last_modified
+                            folders[item_name] = last_modified
+                        else:
+                            size = fileobj.get('Size')
+
+                            # Make size human readable.
+                            human_readadble = ''
```

Comment:
Move this logic into a separate private method.

Reply:
It also looks like the transforming size to human-readable needs to be moved to the view.

Reply:
Maybe it would be more logical to do it in the view, but it is more convenient as it is now. I'll leave it for now.

Reply:
No, you shouldn't do that, because there is a mixture of data and the way it is displayed. I don't understand what are the difficulties to transfer this logic to the view.

Reply:
Ok, done.